code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AllocationAttempt.java

```diff
@@ -136,4 +138,12 @@ public void setPoolIds(Map<Long, Set<Long>> poolIds) {
         this.poolIds = poolIds;
     }
 
+    public Map<String, Object> getAllocatedIPs() {
+        return allocatedIPs;
```

Comment:
Remind that based on our conversation regarding the bug in the scheduler, you need to make this a list.